api/src/main/java/bio/terra/tanagra/service/databaseaccess/CellValue.java

```diff
@@ -0,0 +1,34 @@
+package bio.terra.tanagra.service.databaseaccess;
+
+import bio.terra.tanagra.service.search.DataType;
+
+/**
+ * An interface for the value of a cell within a row within a result table.
+ *
+ * <p>This interface allows us to read data from different databases in a simple but uniform way.
+ * Different database types should implement this for returning values.
+ */
+public interface CellValue {
+
+  /** The type of data in this cell. */
+  DataType dataType();
+
+  /** Returns whether the cell's value is null. */
+  boolean isNull();
```

Comment:
Any consideration towards using optional in the various interfaces rather having the `isNull` field? I'm not sure I have a strong opinion.

Reply:
I admit I didn't think much about that and was just looking at BQ's FieldValue. There's also returning `@Nullable Long/String/etc`.
I think I like optionals most slightly, going with that, thanks.